apps/ts.c

```diff
@@ -538,15 +538,17 @@ static int create_digest(BIO *input, const char *digest, const EVP_MD *md,
 
         *md_value = OPENSSL_hexstr2buf(digest, &digest_len);
         if (*md_value == NULL || md_value_len != digest_len) {
-            OPENSSL_free(*md_value);
-            *md_value = NULL;
             BIO_printf(bio_err, "bad digest, %d bytes "
                        "must be specified\n", md_value_len);
-            return 0;
+            goto err;
         }
     }
     rv = md_value_len;
  err:
+    if (!rv) {
```

Comment:
This is not good practice if rv is an int that could be -1.

Reply:
Thanks.

I have added a new commit to deal with this problem. I found that the the caller create_query() may fail to catch the negative error codes since it only considers 0 as an error code.